Source/AwsCommonRuntimeKit/sdkutils/endpoint/EndpointsRequestContext.swift

```diff
@@ -48,6 +48,34 @@ public class EndpointsRequestContext {
         }
     }
 
+    private func withByteCursorArrayFromStringArray<R>(
+        _ arg: [String], _ body: (UnsafeMutablePointer<aws_byte_cursor>, Int) -> R) -> R {
+        withArrayOfCStrings(arg) { cStrArr in
+            var cursors = cStrArr.map { aws_byte_cursor_from_c_str($0) }
+            let len = cursors.count
+            return cursors.withUnsafeMutableBufferPointer { cursorsPtr in 
+                return body(cursorsPtr.baseAddress!, len)
+            }
```

Comment:
Do we need a mutable pointer to these, or should we update our C API to take a const ptr?